### PR TITLE
use client-id header for device header and cookie id

### PR DESCRIPTION
**Pull request by lluft**

### CODE REVIEW

File: lib/Castle/Request.php [hunk 1]
```diff
@@ -75,26 +75,14 @@ public static function getUserAgent()
...
-    // If the cookie is specifically not defined or not set in the cookie jar
-    // we'll return the special value '?'. This doesn't have any effect on
-    // functionality.
-    if (empty($cid)) {
-      $cid = '?';
+    if (array_key_exists('HTTP_X_CASTLE_DEVICE_ID', $_SERVER)) {
+      return $_SERVER['HTTP_X_CASTLE_DEVICE_ID'];
+    } else {
+      $cid = Castle::getCookieStore()->read('__cid');
+      return empty($cid) ? '' : $cid;
```
Comment (wallin) [inline lib/Castle/Request.php]: This logic need to be kept in order for cURL to not strip the header. See previous PR by @nullaus

**Review by wallin — CHANGES_REQUESTED** [review]

**Comment by lluft** [comment]
@wallin addressed your comment

File: lib/Castle/CookieStore.php [hunk 2]
```diff
@@ -19,12 +20,15 @@ public function destroy($key)
     }
   }
 
+  public function hasKey($key)
+  {
+    return isset($_COOKIE) && array_key_exists($key, $_COOKIE);
+    return true;
```
Comment (wallin) [inline lib/Castle/CookieStore.php]: This doesn't seem necessary

File: test/CookieStore.php [hunk 3]
```diff
@@ -15,12 +16,15 @@ public function destroy($key)
     }
   }
 
+  public function hasKey($key)
+  {
+    return isset($_COOKIE) && array_key_exists($key, $_COOKIE);
+    return true;
```
Comment (wallin) [inline test/CookieStore.php]: Same here

**Review by wallin — CHANGES_REQUESTED** [review]

**Comment by lluft** [comment]
@wallin those were debugging leftovers. Good catch!

**Pull request merged**